pallets/messages/src/types.rs

```diff
@@ -32,13 +32,15 @@ where
 	pub fn map_to_response<BlockNumber>(
 		&self,
 		block_number: BlockNumber,
+		payload_length: u32,
 	) -> MessageResponse<BlockNumber> {
 		MessageResponse {
 			provider_msa_id: self.provider_msa_id,
 			index: self.index,
 			msa_id: self.msa_id,
 			block_number,
 			payload: self.payload.clone().into_inner(),
+			payload_length,
```

Comment:
This `map_to_response` is missing a few things:

1. Payload is wrong for ipfs messages.
2. Since it is always encoding a single schema, we don't need to have payload_length in here for non-ipfs messages.
3. We can also then get better keys. So instead of payload, we should have `cid` for ipfs schemas.
4. (not this story, but) We actually can do one better still for onchain messages in that the payload should be decoded and passed as a json object under payload as well. Since this is an off-chain interaction.

Reply:
I will modify for 1-3 in this PR. I like option 4 more